AMQNET-638 Add SourceLink support

lrushl opened the pull request:
I'd like to add SourceLink support to this library in order to simplify debugging from referencing projects.


Please take a look into the small demo of SourceLink in action 

![0YU7WcwFXa](https://user-images.githubusercontent.com/290543/78409972-cc2a4580-7613-11ea-848f-22b6ceb501c4.gif)

For more details on how to enable debugging with SourceLink please check the following guides:

- [How to Configure Visual Studio to Use SourceLink to Step into NuGet Package Source](http://www.aaronstannard.com/visual-studio-sourcelink-setup/)
- [Debugging into NuGet packages with SourceLink in Visual Studio for Mac](https://docs.microsoft.com/en-us/visualstudio/mac/source-link?view=vsmac-2019)


For more details on SoureLink, please navigate the following links:

- [dotnet/sourcelink](https://github.com/dotnet/sourcelink)
- [SourceLink file specification](https://github.com/dotnet/designs/blob/master/accepted/2020/diagnostics/source-link.md#source-link-file-specification)

michaelandrepearce commented:
Before this is looked into any further, can you provide details on licensing of the dependency.

lrushl commented:
It uses [Apache License 2.0](https://github.com/dotnet/sourcelink/blob/master/License.txt)

Havret commented on src/nms-api/nms-api.csproj:
```diff
@@ -47,6 +47,24 @@ with the License.  You may obtain a copy of the License at
+  <!-- SourceLink Start -->
+  <PropertyGroup>
+    <!-- Publish the repository URL in the built .nupkg (in the NuSpec <Repository> element) -->
+    <PublishRepositoryUrl>true</PublishRepositoryUrl>
```
Not sure about this line, as we already have link to the repository in the package via `RepositoryUrl` tag.

lrushl replied:
According to the documentation [PublishRepositoryUrl](https://github.com/dotnet/sourcelink/blob/master/docs/README.md#publishrepositoryurl) modifies the same `RepositoryUrl` property. The only difference is that it does this dynamically.

I agree that there is no sense to have them both.
I'd rather remove `PublishRepositoryUrl` and do not touch the existing one (with explicit value). 

@Havret , please share your thoughts on this.

Havret replied:
Sounds good to me. 👍

michaelandrepearce commented:
Lgtm. Do you plan to add as well to the openwire and amqp clients?

Havret commented:
@lrushl Could you please file an issue to our [JIRA](https://issues.apache.org/jira/browse/AMQNET-631?jql=project%20%3D%20AMQNET%20AND%20component%20%3D%20NMS) and squash your commits and name the result according to the convention of the repository?

lrushl commented:
> Lgtm. Do you plan to add as well to the openwire and amqp clients?

Thank you. 
Yeah, I do have a plan to add SourceLink support to the [activemq-nms-amqp](https://github.com/apache/activemq-nms-amqp). 

Regarding the [activemq-nms-openwire](https://github.com/apache/activemq-nms-openwire), I'll first try to help with existing PR [[AMQNET-565] - .net standard conversion, project reorganization](https://github.com/apache/activemq-nms-openwire/pull/9) and after that will add SourceLink support in a separate pr.

> @lrushl Could you please file an issue to our JIRA and squash your commits and name the result according to the convention of the repository?

@Havret , sure I'll squash the commits. Regarding the naming convention, could you please point me where I can read more on conventions in this repo (looks like I missed this).

Regarding the JIRA ticket is there any specific conventions too?

Thanks in advance!

michaelandrepearce commented:
@lrushl there isnt any conventions, simply go over to jira, raise a new jira associating it to the correct component, once you have the JIRA, the main thing is you update the commit message so it includes the JIRA if you look at commit history you will see the commit message format with the JIRA prefix.

michaelandrepearce commented:
> Regarding the activemq-nms-openwire, I'll first try to help with existing PR [AMQNET-565] - .net standard conversion, project reorganization and after that will add SourceLink support in a separate pr.

That would be really welcomed, its been something that keeps getting picked up and never quite getting over the finish post, gladly welcome efforts there.

lrushl commented:
Looks like I've addressed all review comments in this pr.

@michaelandrepearce  and @Havret, thank you for the guidance and feedback provided.

Please let me know in case of any other questions or suggestions.

michaelandrepearce commented:
@lrushl what sort of time frame you think until you make some headway on the openwire client? Just so we can work out time frames to get this released for that, currently nothing else is pressing for this to release you see.
